pandas/core/frame.py

```diff
@@ -3822,6 +3822,15 @@ def isetitem(self, loc, value) -> None:
         In cases where `frame.columns` is unique, this is equivalent to
         `frame[frame.columns[i]] = value`.
         """
+        if isinstance(value, DataFrame):
+            if is_scalar(loc):
+                loc = [loc]
+
+            for i, idx in enumerate(loc):
+                arraylike = self._sanitize_column(value.iloc[:, i])
+                self._iset_item_mgr(idx, arraylike, inplace=False)
```

Comment:
in the tests you've added, `i` and `idx` are always the same - is it possible to add a test where `i` and `idx` differ?

Reply:
as in, if I changed this to
```
                arraylike = self._sanitize_column(value.iloc[:, i])
                self._iset_item_mgr(i, arraylike, inplace=False)
```
then there should be a test which fails

Reply:
Good catch, adjusted one of the tests